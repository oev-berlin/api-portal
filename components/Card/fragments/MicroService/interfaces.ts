interface MicroServiceProps {
    serviceName: string,
    isMicroservice: boolean,
}

export type { MicroServiceProps };
