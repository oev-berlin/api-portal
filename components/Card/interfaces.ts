import { ProjectData } from '../../utils/interfaces';

interface CardProps{
    projectData: ProjectData
}

export type { CardProps };
