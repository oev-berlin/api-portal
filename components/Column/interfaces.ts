import {projectData} from "../../utils/interfaces";

interface columnData {
    name: string,
    projects: projectData[],
}

export {columnData};