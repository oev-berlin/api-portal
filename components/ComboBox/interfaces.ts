export interface ComboBoxProps{
    searchOptions: string[],
}
