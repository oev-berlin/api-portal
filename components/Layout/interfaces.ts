import React from 'react';

export interface LayoutProps {
    searchOptions: string[],
    children: React.ReactNode,
}
