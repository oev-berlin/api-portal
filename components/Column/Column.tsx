import {Grid} from "@mui/material";
import {Card} from "../Card";
import {ColumnContainer, ColumnTitle} from "./styles";
import {columnData} from "./interfaces";
import {projectData} from "../../utils/interfaces";

export const Column = ({projects,name}:columnData ) => {

    return (
        <ColumnContainer xs={12} md={6} item container>
            <Grid xs={10} item>
                <ColumnTitle>
                    {name}
                </ColumnTitle>
            </Grid>
            <Grid item xs={10}>
                {
                    projects.map((project:projectData) =>
                        (
                            <Card
                                projectData={project}
                                key={project.id}
                            />
                        )
                    )
                }
            </Grid>
        </ColumnContainer>);
}