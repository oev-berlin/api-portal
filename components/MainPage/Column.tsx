import {Grid, Typography} from "@mui/material";
import {ProjectCard} from "./ProjectCard";
import {columnData} from "./interfaces";

export const Column = (columnData: columnData) => {
    return(
<Grid item container justifyContent={"center"} xs={5.5} border={1}>
    <Grid item xs={10} sx={{p: 2, mt: 2}}>
        <Typography variant={"h5"} textAlign={"center"}>
            {columnData.name}
        </Typography>
    </Grid>
    <Grid xs={10}>
        {
            columnData.projects.map((project) =>
                (
                    <ProjectCard
                        projectData={project}
                    />
                )
            )
        }

    </Grid>
</Grid>);
}