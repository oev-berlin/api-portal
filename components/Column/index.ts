import { Column } from './Column';
import { ColumnProps } from './interfaces';

export { Column };
export type { ColumnProps };
