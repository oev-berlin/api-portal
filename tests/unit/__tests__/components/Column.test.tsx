import { render, cleanup } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';
import React from 'react';
import { Column, ColumnProps } from '../../../../components/Column';
import { ProjectData } from '../../../../utils/interfaces';
// eslint-disable-next-line import/extensions
import 'jest-styled-components';
import { createProjectData } from '../../../../utils/testUtilities';

expect.extend(toHaveNoViolations);

const testProject1: ProjectData = createProjectData({
  title: 'Test Project1',
});
const testProject2: ProjectData = createProjectData({
  title: 'Test Project2',
});

const setupComponent = ({ projects, name }: ColumnProps) => render(
  <Column
    projects={projects}
    name={name}
  />,
);

describe('Column', () => {
  afterEach(() => {
    cleanup();
  });

  describe('UI Tests', () => {
    it('Column component (column name and projects are not empty) should render without crashing', async () => {
      const screen = setupComponent({
        projects: [testProject1, testProject2],
        name: 'microservices',
      });
      expect(screen.container).toBeDefined();
    });
    it('Column component (column name is empty) should render without crashing', async () => {
      const screen = setupComponent({
        projects: [testProject1, testProject2],
        name: '',
      });
      expect(screen.container).toBeDefined();
    });
    it('Column component (column projects is empty) should render without crashing', async () => {
      const screen = setupComponent({
        projects: [],
        name: 'microservices',
      });
      expect(screen.container).toBeDefined();
    });

    it('Column component should render correctly', async () => {
      const screen = setupComponent({ projects: [testProject1, testProject2], name: 'microservices' });
      await screen
        .findByText(/microservices/)
        .then((element) => expect(element).toBeInTheDocument());
      await screen
        .findByText(/Test Project1/)
        .then((element) => expect(element).toBeInTheDocument());
      await screen
        .findByText(/Test Project2/)
        .then((element) => expect(element).toBeInTheDocument());
    });
  });

  describe('Snapshots Tests', () => {
    it('should match a basic snapshot (column name and projects are not empty)', () => {
      const { container } = setupComponent({ projects: [testProject1, testProject2], name: 'microservices' });
      expect(container).toMatchSnapshot();
    });
    it('should match a basic snapshot (column name is empty)', () => {
      const { container } = setupComponent({ projects: [testProject1, testProject2], name: '' });
      expect(container).toMatchSnapshot();
    });
    it('should match a basic snapshot (column projects is empty)', () => {
      const { container } = setupComponent({ projects: [], name: 'microservices' });
      expect(container).toMatchSnapshot();
    });
  });
  describe('Accessibility Tests', () => {
    it('should pass a basic accessibility test', async () => {
      const { container } = setupComponent({ projects: [testProject1, testProject2], name: 'microservices' });
      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });
  });
});
