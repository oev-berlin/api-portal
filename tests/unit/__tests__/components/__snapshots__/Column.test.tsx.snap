// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column Snapshots Tests should match a basic snapshot (column name and projects are not empty) 1`] = `
.c5 {
  text-align: center;
  padding: 0.5rem;
}

.c3 {
  text-align: left;
  padding-left: 0.5rem;
  text-transform: none;
}

.c2 {
  margin-bottom: 1rem;
}

.c4 {
  padding: 1rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 {
  margin-top: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: solid;
}

.c1 {
  margin: 1rem;
  text-align: center;
}

<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 c0 css-18kb45e-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
    >
      <h2
        class="c1"
      >
        microservices
      </h2>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root c2 css-bhp9pd-MuiPaper-root-MuiCard-root"
        href="/project/Test%20Project1"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root MuiCardActionArea-root css-1hu6zam-MuiButtonBase-root-MuiButton-root-MuiButtonBase-root-MuiCardActionArea-root"
          role="button"
          tabindex="0"
          type="button"
        >
          <h3
            class="c3"
          >
            Test Project1
          </h3>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 c4 css-mhc70k-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c5 css-1ps6pg7-MuiPaper-root"
              >
                PDF Service
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c5 css-1ps6pg7-MuiPaper-root"
              >
                Email Service
              </div>
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root c2 css-bhp9pd-MuiPaper-root-MuiCard-root"
        href="/project/Test%20Project2"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root MuiCardActionArea-root css-1hu6zam-MuiButtonBase-root-MuiButton-root-MuiButtonBase-root-MuiCardActionArea-root"
          role="button"
          tabindex="0"
          type="button"
        >
          <h3
            class="c3"
          >
            Test Project2
          </h3>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 c4 css-mhc70k-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c5 css-1ps6pg7-MuiPaper-root"
              >
                PDF Service
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c5 css-1ps6pg7-MuiPaper-root"
              >
                Email Service
              </div>
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Column Snapshots Tests should match a basic snapshot (column name is empty) 1`] = `
.c5 {
  text-align: center;
  padding: 0.5rem;
}

.c3 {
  text-align: left;
  padding-left: 0.5rem;
  text-transform: none;
}

.c2 {
  margin-bottom: 1rem;
}

.c4 {
  padding: 1rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 {
  margin-top: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: solid;
}

.c1 {
  margin: 1rem;
  text-align: center;
}

<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 c0 css-18kb45e-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
    >
      <h2
        class="c1"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root c2 css-bhp9pd-MuiPaper-root-MuiCard-root"
        href="/project/Test%20Project1"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root MuiCardActionArea-root css-1hu6zam-MuiButtonBase-root-MuiButton-root-MuiButtonBase-root-MuiCardActionArea-root"
          role="button"
          tabindex="0"
          type="button"
        >
          <h3
            class="c3"
          >
            Test Project1
          </h3>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 c4 css-mhc70k-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c5 css-1ps6pg7-MuiPaper-root"
              >
                PDF Service
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c5 css-1ps6pg7-MuiPaper-root"
              >
                Email Service
              </div>
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root c2 css-bhp9pd-MuiPaper-root-MuiCard-root"
        href="/project/Test%20Project2"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root MuiCardActionArea-root css-1hu6zam-MuiButtonBase-root-MuiButton-root-MuiButtonBase-root-MuiCardActionArea-root"
          role="button"
          tabindex="0"
          type="button"
        >
          <h3
            class="c3"
          >
            Test Project2
          </h3>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 c4 css-mhc70k-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c5 css-1ps6pg7-MuiPaper-root"
              >
                PDF Service
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c5 css-1ps6pg7-MuiPaper-root"
              >
                Email Service
              </div>
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Column Snapshots Tests should match a basic snapshot (column projects is empty) 1`] = `
.c0 {
  margin-top: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: solid;
}

.c1 {
  margin: 1rem;
  text-align: center;
}

<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 c0 css-18kb45e-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
    >
      <h2
        class="c1"
      >
        microservices
      </h2>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
    />
  </div>
</div>
`;
